### ISSUE TITLE
Compilation failing due to a warning

### ISSUE THREAD

**Issue by karouf** [Shopify/liquid-c#20]
The `liquid-c` gem is bundled in [Jekyll](https://github.com/jekyll/jekyll) and when I updated my bundle earlier on it updated `liquid-c` from 0.0.2 to 0.0.3. It failed during the compilation. I tried again installing only the gem, it also failed with the following log:

```
$ gem install liquid-c -v '0.0.3'


Building native extensions.  This could take a while...
ERROR:  Error installing liquid-c:
	ERROR: Failed to build gem native extension.

    /Users/kara/.rvm/rubies/ruby-2.0.0-p353/bin/ruby extconf.rb
creating Makefile

make "DESTDIR=" clean

make "DESTDIR="
compiling block.c
compiling lexer.c
compiling liquid.c
compiling parser.c
cc1: warnings being treated as errors
parser.c: In function ‘rb_parse_expression’:
parser.c:165: warning: unused parameter ‘self’
make: *** [parser.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/kara/.rvm/gems/ruby-2.0.0-p353/gems/liquid-c-0.0.3 for inspection.
Results logged to /Users/kara/.rvm/gems/ruby-2.0.0-p353/extensions/x86_64-darwin-10/2.0.0/liquid-c-0.0.3/gem_make.out
```

I also tried to install v0.0.2 with no problems.

I don't know if it has anything to do with my setup but I managed to get past this by installing `liquid-c` by itself with the `-w` cflag in order to ignore warnings during compilation:

```
$ gem install liquid-c -v '0.0.3' -- --with-cflags=-w
```

It can probably be corrected with a modification to [ext/liquid_c/extconf.rb](https://github.com/Shopify/liquid-c/blob/master/ext/liquid_c/extconf.rb)

**Comment by pushrax** [comment]
Thanks for the report. Looks like your compiler is doing -Wextra or similar by default. #21 should disable that warning.

**Comment by pushrax** [comment]
Fix released in 3.0.0
